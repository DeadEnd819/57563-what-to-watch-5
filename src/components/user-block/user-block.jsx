import React from "react";
import PropTypes from "prop-types";
import {Link} from "react-router-dom";
import {connect} from "react-redux";
import {isUserLoggedIn, getUserAvatar} from "../../store/selectors";
import {AppRoute} from "../../const";

const UserBlock = ({isUserLogged, avatarUrl}) => {
  const {LOGIN, MY_LIST} = AppRoute;

  const getUserBlockElement = () => {
    if (!isUserLogged) {
      return (<Link className="user-block__link" to = {LOGIN}>Sign in</Link>);
    }

    return (
      <div className="user-block__avatar">
        <Link to = {MY_LIST}>
          <img src={avatarUrl} alt="User avatar" width="63" height="63" />
        </Link>
      </div>
    );
  };

  return (
    <div className="user-block">
      {getUserBlockElement()}
    </div>
  );
};

UserBlock.propTypes = {
  isUserLogged: PropTypes.bool.isRequired,
  avatarUrl: PropTypes.string,
};

const mapStateToProps = (state) => ({
  isUserLogged: isUserLoggedIn(state),
  avatarUrl: getUserAvatar(state),
});

export {UserBlock};
export default connect(mapStateToProps)(UserBlock);

