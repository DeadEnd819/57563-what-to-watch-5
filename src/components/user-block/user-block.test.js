import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import {UserBlock} from "./user-block";

const renderer = new ShallowRenderer();

describe(`Render UserBlock`, () => {
  it(`Should UserBlock render correctly`, () => {
    renderer.render(
        <UserBlock isUserLogged={false} avatarUrl={`avatar.jpg`} />
    );

    const tree = renderer.getRenderOutput();

    expect(tree).toMatchSnapshot();
  });

  it(`Should UserBlock logged render correctly`, () => {
    renderer.render(
        <UserBlock isUserLogged={true} avatarUrl={`avatar.jpg`} />
    );

    const tree = renderer.getRenderOutput();

    expect(tree).toMatchSnapshot();
  });
});
