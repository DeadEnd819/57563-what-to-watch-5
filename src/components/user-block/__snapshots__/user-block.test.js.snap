// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render UserBlock Should UserBlock logged render correctly 1`] = `
<div
  className="user-block"
>
  <div
    className="user-block__avatar"
  >
    <Link
      to="/mylist"
    >
      <img
        alt="User avatar"
        height="63"
        src="avatar.jpg"
        width="63"
      />
    </Link>
  </div>
</div>
`;

exports[`Render UserBlock Should UserBlock render correctly 1`] = `
<div
  className="user-block"
>
  <Link
    className="user-block__link"
    to="/login"
  >
    Sign in
  </Link>
</div>
`;
